source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -130,12 +130,19 @@ bool InstanceImplPosix::illegalPath(const std::string& path) {
   // instances poking in bad places. We may have to consider conditioning on
   // platform in the future, growing these or relaxing some constraints (e.g.
   // there are valid reasons to go via /proc for file paths).
-  // TODO(htuch): Optimize this as a hash lookup if we grow any further.
-  if (absl::StartsWith(canonical_path.rc_, "/dev") ||
-      absl::StartsWith(canonical_path.rc_, "/sys") ||
-      absl::StartsWith(canonical_path.rc_, "/proc")) {
-    return true;
+  auto paths = []() {
+    CONSTRUCT_ON_FIRST_USE(std::vector<std::string>, {"/dev", "/sys", "/proc"});
```

Comment:
How to deal with the path like "/dev/fd/../../develop"? 
maybe it will be added to whitelist by line 118.

Reply:
see line 122, `canonicalPath(path);` can deal with that.

Reply:
The path will be checked by StartWith() function in line 118, and return false directly.
Could you pls check again?

Reply:
I guess you want to say this method should return true(illegal) for `/dev/fd/../../dev`, but line 118 makes it return false(legal), right? 
The code from line 111 to line 120 is a hack for special case, it's commented verbosely, I don't think it will cause any issue in reality. No one call this method with path like `/dev/fd/../../`.

Reply:
Got it, thanks for your explanation.